src/emulator/cpu.ts

```diff
@@ -282,12 +300,20 @@ export class Cpu {
                 return instruction.cycles;
             }
 
-            case Operation.jp:
-                this.clock.increment(instruction.cycles);
+            case Operation.jp: {
+                const condition = this.evaluateCondition(instruction);
+                const cycles = instruction.cycles + (condition ? 1 : 0);
```

Comment:
If this is to hold for unconditional `JP`, then the cycle count in the instruction defintion needs to be adjusted.